src/components/MdnRequest.vue

```diff
@@ -0,0 +1,61 @@
+<template>
+	<div v-if="hasMdnRequest && !mdnSent" class="mail-message-has-mdn-request">
+		{{
+			t('mail', 'The sender of this message has asked to be notified when you read this message.')
+		}}
+		<button :disabled="loading" @click="sendMdn">
+			{{ t('mail', 'Notify the sender') }}
+		</button>
+	</div>
+	<div v-else-if="mdnSent" class="mail-message-has-mdn-request">
+		{{
+			t('mail', 'You sent a read confirmation to the sender of this message.')
+		}}
+	</div>
+</template>
+
+<script>
+import logger from '../logger'
+import { sendMdn } from '../service/MessageService'
+import { showError } from '@nextcloud/dialogs'
+
+export default {
+	name: 'MdnRequest',
+	props: {
+		message: {
+			required: true,
+			type: Object,
+		},
+	},
+	data() {
+		return {
+			mdnSent: this.message.flags.mdnsent,
```

Comment:
oh and this should be moved into a *computed* property so it becomes reactive and updates when the flag changes